packages/editor/src/components/post-permalink/index.js

```diff
@@ -78,25 +103,35 @@ class PostPermalink extends Component {
 
 				<span className="editor-post-permalink__label">{ __( 'Permalink:' ) }</span>
 
-				{ ! isEditingPermalink &&
+				{ ! isEditingPermalink && isPublished &&
 					<ExternalLink
 						className="editor-post-permalink__link"
 						href={ ! isPublished ? postLink : samplePermalink }
 						target="_blank"
 						ref={ ( linkElement ) => this.linkElement = linkElement }
 					>
-						{ safeDecodeURI( samplePermalink ) }
+						{ ( currentTitle || currentSlug ) ? safeDecodeURI( samplePermalink ) : postLink }
 						&lrm;
 					</ExternalLink>
 				}
 
+				{ ! isEditingPermalink && ! isPublished &&
+					<PostPreviewButton
+						isPermalink={ true }
+						className="editor-post-permalink__link"
+						samplePermalink={ samplePermalink }
+						linkElement={ ( linkElement ) => this.linkElement = linkElement }
+					/>
+				}
+
 				{ isEditingPermalink &&
 					<PostPermalinkEditor
+						postSlug={ slug }
 						onSave={ () => this.setState( { isEditingPermalink: false } ) }
 					/>
 				}
 
-				{ isEditable && ! isEditingPermalink &&
+				{ isEditable && ! isEditingPermalink && ! isNew &&
```

Comment:
Can we have the same `! isNew` check for the `Change Permalinks` button? That shouldn't be displayed for new posts.

Reply:
Now that I look at this again, this isn't actually doing anything, so I could drop this check. The entire component returns `null` on render if `isNew` is true ([See line 82](https://github.com/WordPress/gutenberg/pull/12009/files#diff-21ca0f66cf1102449de446bf6bf2fed2R82)), so it's always false at this point.